services/data_fetch_service.py

```
@@ -86,17 +86,34 @@ def _compute_dataset_hash(df: pd.DataFrame | None) -> str:
+    collected: list[str] = []
+    seen: set[str] = set()
+
+    for value in df["tipo"].tolist():
+        if value in (None, ""):
+            continue
+        if isinstance(value, float):
+            try:
+                if pd.isna(value):
+                    continue
+            except Exception:
+                pass
```

Ruff (0.14.1)

`try`-`except`-`pass` detected, consider logging the exception

(S110)